app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -91,55 +52,103 @@ def perform(_now)
+    end
+  end
 
-      status_check_attempted_at = Time.zone.now
-      enrollment_outcomes[:enrollments_checked] += 1
-      response = nil
+  def check_enrollment(enrollment, is_not_last_enrollment)
+    # Add a unique ID for enrollments that don't have one
+    enrollment.update(unique_id: enrollment.usps_unique_id) if enrollment.unique_id.blank?
 
-      begin
-        response = proofer.request_proofing_results(
-          enrollment.unique_id, enrollment.enrollment_code
-        )
-      rescue Faraday::BadRequestError => err
-        # 400 status code. This is used for some status updates and some common client errors
-        handle_bad_request_error(err, enrollment)
-      rescue Faraday::ClientError, Faraday::ServerError => err
-        # 4xx or 5xx status code. These are unexpected but will have some sort of
-        # response body that we can try to log data from
-        handle_client_or_server_error(err, enrollment)
-      rescue Faraday::Error => err
-        # Timeouts, failed connections, parsing errors, and other HTTP errors. These
-        # generally won't have a response body
-        handle_faraday_error(err, enrollment)
-      rescue StandardError => err
-        handle_standard_error(err, enrollment)
-      else
-        process_enrollment_response(enrollment, response)
-      ensure
-        # Record the attempt to update the enrollment
-        enrollment.update(status_check_attempted_at: status_check_attempted_at)
-      end
+    status_check_attempted_at = Time.zone.now
+    enrollment_outcomes[:enrollments_checked] += 1
+    response = nil
 
-      # Sleep for a while before we attempt to make another call to USPS
-      sleep request_delay_in_seconds if idx < enrollments.length - 1
+    begin
+      response = proofer.request_proofing_results(
+        enrollment.unique_id, enrollment.enrollment_code
+      )
+    rescue Faraday::BadRequestError => err
+      # 400 status code. This is used for some status updates and some common client errors
+      handle_bad_request_error(err, enrollment)
+    rescue Faraday::ClientError, Faraday::ServerError => err
+      # 4xx or 5xx status code. These are unexpected but will have some sort of
+      # response body that we can try to log data from
+      handle_client_or_server_error(err, enrollment)
+    rescue Faraday::Error => err
+      # Timeouts, failed connections, parsing errors, and other HTTP errors. These
+      # generally won't have a response body
+      handle_faraday_error(err, enrollment)
+    rescue StandardError => err
+      handle_standard_error(err, enrollment)
+    else
+      process_enrollment_response(enrollment, response)
+    ensure
+      # Record the attempt to update the enrollment
+      enrollment.update(status_check_attempted_at: status_check_attempted_at)
     end
+
+    # Sleep for a while before we attempt to make another call to USPS
+    sleep request_delay_in_seconds if is_not_last_enrollment
   end
```

Comment:
one idea: we could move this `sleep` outside of this method (maybe back to the loop) and we could use the implicit begin/rescue for the method and unindent a bunch, I just confirmed top-level `else` is supported

Looks like then we could even drop the `is_not_last_enrollment` arg entirely

```suggestion
   response = proofer.request_proofing_results(
      enrollment.unique_id, enrollment.enrollment_code
    )
  rescue Faraday::BadRequestError => err
    # 400 status code. This is used for some status updates and some common client errors
    handle_bad_request_error(err, enrollment)
  rescue Faraday::ClientError, Faraday::ServerError => err
    # 4xx or 5xx status code. These are unexpected but will have some sort of
    # response body that we can try to log data from
    handle_client_or_server_error(err, enrollment)
  rescue Faraday::Error => err
    # Timeouts, failed connections, parsing errors, and other HTTP errors. These
    # generally won't have a response body
    handle_faraday_error(err, enrollment)
  rescue StandardError => err
    handle_standard_error(err, enrollment)
  else
    process_enrollment_response(enrollment, response)
  ensure
    # Record the attempt to update the enrollment
    enrollment.update(status_check_attempted_at: status_check_attempted_at)
  end
```